nucypher/blockchain/eth/signers/hardware.py

```diff
@@ -0,0 +1,243 @@
+        return True
+
+    @validate_checksum_address
+    def lock_account(self, account: str) -> bool:
+        """Defer account locking to the trezor, do not indicate application level unlocking logic."""
+        return True
+
+    @property
+    def accounts(self) -> List[str]:
+        """Returns a list of cached trezor checksum addresses from initial derivation."""
+        return list(self.__addresses.keys())
+
+    @handle_trezor_call
+    def sign_message(self, message: bytes, checksum_address: str) -> SignedMessage:
+        """
+        Signs a message via the TREZOR ethereum sign_message API and returns
+        a named tuple containing the signature and the address used to sign it.
+        This method requires interaction between the TREZOR and the user.
+        """
+        hd_path = self.__get_address_path(checksum_address=checksum_address)
+        signed_message = trezorlib.ethereum.sign_message(self.__client, hd_path, message)
+        return self.SignedMessage(signed_message.signature, signed_message.address)
+
+    def sign_transaction(self,
+                         transaction_dict: dict,
+                         rlp_encoded: bool = True
+                         ) -> Union[HexBytes, Transaction]:
+        """
+        Sign a transaction with a trezor hardware wallet.
+
+        This method handles transaction validation, field formatting, signing,
+        and outgoing serialization.  Accepts a standard transaction dictionary as input,
+        and produces an RLP encoded raw signed transaction by default.
+
+        Internally the standard transaction dictionary is reformatted for trezor API consumption
+        via calls `trezorlib.client.ethereum.sign_tx`.
+
+        WARNING: This function returns a raw signed transaction which can be
+        broadcast by anyone with a connection to the ethereum network.
+
+        ***Treat pre-signed raw transactions produced by this function like money.***
+
+        """
+
+        # Read the sender inside the transaction request
+        checksum_address = transaction_dict.pop('from')
+
+        # Format contract data field for both trezor and eth_account
+        if transaction_dict.get('data') is not None:  # empty string is valid
+            transaction_dict['data'] = Web3.toBytes(HexBytes(transaction_dict['data']))
+
+        # Format transaction fields for Trezor, Lookup HD path, and Sign Transaction
+        # If `chain_id` is included, an EIP-155 transaction signature will be applied
+        # https://github.com/trezor/trezor-core/pull/311
+        trezor_transaction = self._format_transaction(transaction_dict=transaction_dict)
+        n = self.__get_address_path(checksum_address=checksum_address)
+        v, r, s = self.__sign_transaction(n=n, trezor_transaction=trezor_transaction)
+
+        # Format the transaction for eth_account Transaction consumption
+        # v = (v + 2) * (chain_id + 35)
```

Comment:
This comment is confusing to me. This formula is nowhere to be seen in the code. Should it be here?

Reply:
Agreed.  I pushed an update, let's make this more clear. :+1: